src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
```

Comment:
To me this feels like this is just an inner function definition in `metrics.js` is there a specific reason it's been moved to a separate file? Right now the coupling between the two feels unnecessarily high.